src/Framework/Sdk/SdkResultItem.cs

```diff
@@ -55,14 +60,14 @@ public override bool Equals(object obj)
         public override int GetHashCode()
         {
             int hashCode = -849885975;
-            hashCode = (hashCode * -1521134295) + EqualityComparer<string>.Default.GetHashCode(ItemSpec);
+            hashCode = hashCode ^ ItemSpec.GetHashCode();
 
-            if (Metadata != null)
+            if (Metadata != null && Metadata.Count > 0)
```

Comment:
I don't think we really need to check `Count` here; `foreach` over an empty list is fast.

Also, `Metadata` is never null, right?

Reply:
It can be if you pass in null as the metadata when constructing it.

Reply:
Done removing unnecessary Count check.
I've added null check too, because the other constructor may accept Metadata parameter as null.